Examples/Tests/collision/analysis_collision_2d.py

```diff
@@ -74,6 +76,30 @@
 print('tolerance = ', tolerance)
 assert(error < tolerance)
 
-fn = fn_list[-1]
-test_name = fn[:-9] # Could also be os.path.split(os.getcwd())[1]
+
+## In the second past of the test, we verify that the diagnostic particle filter function works as
+## expected. For this, we only use the last simulation timestep.
+
+last_fn_filtered = last_fn.replace("diag1", "diag_filter")
+ds  = yt.load( last_fn )
+ds_filtered  = yt.load( last_fn_filtered )
+ad  = ds.all_data()
+ad_filtered  = ds_filtered.all_data()
+
+# Load data from the non-filtered diagnostic. [:ne] is here so that we discard the ions, which are
+# not dumped by the filtered diag
+ids = ad['particle_id'].to_ndarray()[:ne]
+x = ad['particle_position_x'].to_ndarray()[:ne]
+px  = ad['particle_momentum_x'].to_ndarray()[:ne]
+z = ad['particle_position_y'].to_ndarray()[:ne]
+pz  = ad['particle_momentum_z'].to_ndarray()[:ne]
```

Comment:
This can be quite tricky for users, doesn't it? Is this well-documented? I did not see anything mentioning this in the page _Visualization with yt (for plotfiles)_ of the documentation.

Reply:
cc @Yin-YinjianZhao for help.

Reply:
@NeilZaim You meant the 'position_y' to z?

Reply:
@Yin-YinjianZhao Yes!